File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // Get user cash amount
    float dollars;
    do
    {
        dollars = get_float("Enter dollar amount:\n");
    }
    while (dollars <= 0);
    
    // Convert to cents
    int cents = round(dollars * 100);
    
    // Use while loop to iterate through change
    int numCoins = 0;
    
    // Now calculate change
    while (cents >= 25)
    {
        cents -= 25;
        numCoins++;
    }
    while (cents >= 10)
    {
        cents -= 10;
        numCoins++;
    }
    while (cents >= 5)
    {
        cents -= 5;
        numCoins++;
    }
    while (cents >= 1)
    {
        cents -= 1;
        numCoins++;
    }
    
    // output entered cash amount
    printf("%i\n",numCoins);
}

File: pset6/sentimental-cash/cash.py
from cs50 import get_float

def get_money():
    while True:
        n = get_float("Change owed: ")
        if n > 0:
            break
    return n

change = get_money() * 100
numCoins = 0

while change >= 25:
    change -= 25
    numCoins += 1;
while change >= 10:
    change -= 10
    numCoins += 1
while change >= 5:
    change -= 5
    numCoins += 1
while change >= 1:
    change -= 1
    numCoins += 1

print(f"{numCoins}\n")

